Framework/Algorithms/src/LineProfile.cpp

```diff
@@ -87,6 +89,39 @@ struct IndexLimits {
   size_t widthEnd;
 };
 
+/**
+ * Create the profile workspace.
+ * @param ws The parent workspace.
+ * @param Xs Profile's X values.
+ * @param Ys Profile's Y values.
+ * @param Es Profile's E values.
+ * @return A single histogram profile workspace.
+ */
+Workspace2D_sptr makeOutput(const MatrixWorkspace &ws,
+                            const LineDirection direction,
+                            std::vector<double> &&Xs, std::vector<double> &&Ys,
+                            std::vector<double> &&Es) {
+  Workspace2D_sptr outWS;
+  const bool histogramOutput = direction == LineDirection::horizontal
```

Comment:
Have a look at `HistogramBuilder`, I think you might be able to use it to avoid distinguishing all those cases here and below. Just build the histogram and use it with a single `create` call.